wcfsetup/install/files/lib/acp/form/LanguageAddForm.class.php

```diff
@@ -103,149 +75,123 @@ public function readParameters()
+                        ->label('wcf.global.name')
+                        ->description('wcf.acp.language.name.description')
+                        ->maximumLength(255)
+                        ->required(),
+                    TextFormField::create('languageCode')
+                        ->label('wcf.acp.language.code')
+                        ->description('wcf.acp.language.code.description')
+                        ->maximumLength(20)
+                        ->required()
+                        ->addValidator(new FormFieldValidator('unique', function (TextFormField $formField) {
+                            if ($formField->getValidationErrors() !== []) {
+                                return;
+                            }
+
+                            if (
+                                $this->formObject instanceof Language
+                                && \mb_strtolower($this->formObject->languageCode) === $formField->getValue()
+                            ) {
+                                return;
+                            }
+
+                            if (LanguageFactory::getInstance()->getLanguageByCode($formField->getValue())) {
+                                $formField->addValidationError(new FormFieldValidationError(
+                                    'notUnique',
+                                    'wcf.acp.language.add.languageCode.error.notUnique'
+                                ));
+                            }
+                        })),
+                    TextFormField::create('countryCode')
+                        ->label('wcf.acp.language.countryCode')
+                        ->description('wcf.acp.language.countryCode.description')
+                        ->maximumLength(10)
+                        ->required(),
+                    SingleSelectionFormField::create('locale')
+                        ->label('wcf.acp.language.locale')
+                        ->description('wcf.acp.language.locale.description')
+                        ->options($locales)
+                        ->addValidator(new FormFieldValidator(
+                            'locale',
+                            static function (SingleSelectionFormField $formField) {
+                                if ($formField->getValidationErrors() !== []) {
+                                    return;
+                                }
+
+                                $languageCodeField = $formField->getDocument()->getNodeById('languageCode');
+                                \assert($languageCodeField instanceof TextFormField);
+                                if ($languageCodeField->getValue() === '') {
+                                    return;
+                                }
+
+                                $test = \explode('_', $formField->getValue());
+                                if (\count($test) === 1 || $languageCodeField->getValue() !== $test[0]) {
```

Comment:
```suggestion
                                if ($languageCodeField->getValue() !== $test[0]) {
```
Is there a reason you rejected the locales without an actual region code? I feel it should be legal to select “Italian” when create the `it` language, even if it is not all that useful to select just the locale without a region.